docs/dev/vagrant.rst

```diff
@@ -121,13 +130,14 @@ For instructions on contributing changes and our ``git`` workflow, see `<contrib
 Problems
 --------
 
-1. The ``vagrant up`` command errors out with a Ruby stack trace.
+1. The ``vagrant up`` command errors out, or times out while waiting for the VM to boot. (You may also want to investigate some of these for errors later in the process.)
+
+    If it errors out with a Ruby stack trace, there is a `known issue <https://github.com/mitchellh/vagrant/issues/6748>`_ with Vagrant/VirtualBox on IPv6 static networking.
 
-    There is a `known issue <https://github.com/mitchellh/vagrant/issues/6748>`_ with Vagrant/VirtualBox on IPv6 static networking.
+    One other thing to try is to run the VM not headlessly. You can run the VM directly from VirtualBox. You can also do this in Vagrant by uncommenting the line ``# vb.gui = true`` in ``Vagrantfile``, then running ``vagrant reload``. VirtualBox may give a more helpful error message, or you may be able to observe the VM getting stuck waiting for a keypress that never comes, say on the bootloader.
```

Comment:
I believe you may also be able to open it in Virtualbox even if it is already running in headless mode.